### PR TITLE
Use explorer as method

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/hvplot/issues/1153
```python
import hvplot.pandas

from bokeh.sampledata.penguins import data as df

df.hvplot.explorer(title="ABCD", kind="scatter", x="flipper_length_mm", y="body_mass_g", by=["species"])
```

image

### CODE REVIEW

**Review by maximlt — COMMENTED** [review]
hvPlot's main entry points are so far:
- `df.hvplot.<ptype>()`
- `df.interactive()`
- `hvplot.explorer(df)`

The first two approaches are patching the data structures that support their operations, dynamically adding the `hvplot` and `interactive` accessors. For some reason, this wasn't done with `explorer` in the first place. Maybe because the explorer is quite different in nature compared to the other two accessors, being basically a not-so-small app. There's also the possibility (not sure it works!) that you can create an explorer from some dataset and update it with another dataset, maybe from a different datatype entirely. That's actually a valid use case for the explorer, imagine a Lumen app filtering/transforming some data that ends up in a pre-defined explorer for more exploration. The more common use case though for the explorer is to operate on a single dataset. In that case, the API offered is inconsistent with the two other entry points and  and this is what this PR aims to improve.

One could suggest that hvPlot could add an `explorer` accessor as it does with `hvplot` and `interactive`. I thought about that a little but prefer the approach of this PR, adding a new `explorer` method under `df.hvplot`, as the explorer is very close to the plotting API (a lot more than `.interactive`) and GeoPandas has recently added an `.explore` method (to get an interactive plot), which could be a little confusing for their users.

@ahuang11 there needs to be a few more docs update:
- [x] landing page
- [ ] explorer getting started

And I have left a few comments.

File: examples/user_guide/Explorer.ipynb [hunk 1]
```diff
@@ -6,13 +6,14 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import hvplot.pandas\n",
+    "import hvplot.pandas  # noqa: F401\n",
```
[the remark this answers is not in the document]
Comment (ahuang11) [reply examples/user_guide/Explorer.ipynb]: pre-commit wants it I think.

File: hvplot/plotting/core.py [hunk 2]
```diff
@@ -1814,6 +1815,50 @@ def labels(self, x=None, y=None, text=None, **kwds):
...
+                    "string": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
+                },
+            )
+            df.hvplot.explorer()
+        """
+        explorer_kwds = {"_explorer_kwds": kwds}
+        explorer_kind = kwds.pop("kind", None)
+        if explorer_kind:
+            explorer_kwds["_explorer_kind"] = explorer_kind
+        return self(x, y, kind="explorer", **explorer_kwds)
```
[the remark this answers is not in the document]
Comment (ahuang11) [reply hvplot/plotting/core.py]: Good idea; I overcomplicated it.

File: hvplot/plotting/core.py [hunk 3]
```diff
@@ -1814,6 +1815,50 @@ def labels(self, x=None, y=None, text=None, **kwds):
         """
         return self(x, y, text=text, kind="labels", **kwds)
 
+    def explorer(self, x=None, y=None, **kwds):
```
[the remark this answers is not in the document]
Comment (ahuang11) [reply hvplot/plotting/core.py:1818]: Okay I suppose we can wait until that's done first.

File: examples/user_guide/Explorer.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahuang11) [reply examples/user_guide/Explorer.ipynb]: https://github.com/holoviz/hvplot/actions/runs/6401029977/job/17375503588?pr=1154

[the remark this answers is not in the document]
Comment (maximlt) [reply examples/user_guide/Explorer.ipynb]: Ok that's nbqa, thanks for confirming that. Then just `# noqa` is fine and it's how it's done in other notebooks, `# noqa: F401` is a tad ugly.

File: hvplot/plotting/core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (maximlt) [reply hvplot/plotting/core.py]: I believe you can remove all the code you added in `converter.py` and just touch `core.py`.

**Comment by maximlt** [comment]
I have made these changes:

- moved the `explorer` method to `hvPlotBase`
- replaced explorer handling in converter.py by handling in `hvPlotBase.__call__`; I prefer not to have any explorer code in the HoloViewsConverter, it's complicated enough :D

**Pull request merged**